geps/gep-2648/index.md

```diff
@@ -0,0 +1,465 @@
+* In terms of status, it SHOULD be reasonably easy for a user to understand that
+  everything is working - basically, as long as the targeted object exists,
+  the modifications are valid, the metaresource is valid, and this should be
+  straightforward to communicate in one or two Conditions. The `status` stanza
+  in BackendTLSPolicy is an example of one of the recommended ways to achieve this.
+* Direct Policy Attachment SHOULD only be used to target objects in the same
+  namespace as the Policy object. Allowing cross-namespace references brings in
+  significant security concerns, and/or difficulties about merging cross-namespace
+  policy objects. Notably, Mesh use cases may need to do something like this for
+  consumer policies, but in general, Policy objects that modify the behavior of
+  things outside their own namespace should be avoided unless it uses a handshake
+  of some sort, where the things outside the namespace can opt–out of the behavior.
+  (Notably, this is the design that we used for ReferenceGrant).
+
+## Apply Policies to Sections of a Resource
+
+The `sectionName` field of `targetRef` can be used to target a specific section
+of other resources, for example:
+
+* Service.Ports.Name
+* Gateway.Listeners.Name
+* HTTPRoute.Rules.Name (once they are added in GEP-995, PR at https://github.com/kubernetes-sigs/gateway-api/pull/2593)
+
+For example, the RetryPolicy below applies to a RouteRule inside an HTTPRoute.
+(or rather, it will when GEP-995 merges).
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: HTTPRoute
+metadata:
+  name: http-app-1
+  labels:
+    app: foo
+spec:
+  hostnames:
+  - "foo.com"
+  rules:
+  - name: bar
+    matches:
+    - path:
+        type: Prefix
+        value: /bar
+    backendRefs:
+    - name: my-service1
+      port: 8080
+---
+apiVersion: networking.acme.io/v1alpha2
+kind: RetryPolicy
+metadata:
+  name: foo
+spec:
+  maxRetries: 5
+  targetRef:
+    name: http-app-1
+    group: gateway.networking.k8s.io
+    kind: HTTPRoute
+    sectionName: bar
+```
+
+If a `sectionName` is specified, but does not exist on the targeted object, the
+Policy must fail to attach, and the policy implementation should record a
+`resolvedRefs` failure or similar Condition in the Policy's status.
+
+When multiple Policies of the same type target the same object, one with a
+`sectionName` and one without, the more specific policy (i.e., the one with a
+`sectionName`) will have its entire `spec` applied to the named section.
+The less specific policy will also have its `spec` applied to the target but
+MUST not affect the named section. The less specific policy will have its `spec`
+applied to all other sections of the target that are not targeted by any other
+more specific policies.
+
+When more than one object matches the same object _and_ `sectionName`, the usual
+conflict-resolution rules (as defined in GEP-713 should be used). These boil down
+to "oldest by creation date wins".
```

Comment:
@candita I don't think this clarification actually solves for your use case, where you're explicitly trying to match multiple objects with a given backendRef `name`.

Refs https://github.com/kubernetes-sigs/gateway-api/pull/2813#discussion_r1520616890 and https://github.com/kubernetes-sigs/gateway-api/pull/2813#discussion_r1552455260

Reply:
I added a "SHOULD NOT use `backendRef`" above, which should clear this up, I think.

Reply:
@mikemorris I wanted information on what was considered a `SectionName` in the object.  I [heard from Rob that we have known `SectionNames` for each object.](https://github.com/kubernetes-sigs/gateway-api/pull/2813#discussion_r1554111330)  I do hope we have this documented somewhere.

Reply:
Opened https://github.com/kubernetes-sigs/gateway-api/pull/2946 to improve documentation for SectionName